gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -289,6 +297,9 @@ public Spec getSpecWrapper(URI uri) {
   /**
    * Persist {@link Spec} into {@link SpecStore} and notify {@link SpecCatalogListener} if triggerListener
    * is set to true.
+   * If the {@link Spec} is a {@link FlowSpec} it is persisted if it can be compiled at the time this method received
+   * the spec. `explain` specs are not persisted. The logic of this method is tightly coupled with the logic of
```

Comment:
Javadoc can be enhanced to explain how the coupling is achieved e.g. we use condition variables to achieve synchronization between NonScheduledJobRunner thread and FlowCatalog thread to ensure deletion of FlowSpec happens after the corresponding run once flow is submitted to the orchestrator.